pkg/webhooks/machine_webhook.go

```diff
@@ -1426,3 +1438,14 @@ func validateAzureDataDisks(machineName string, spec *machinev1.AzureMachineProv
 func isDeleting(obj metav1.Object) bool {
 	return obj.GetDeletionTimestamp() != nil
 }
+
+// isFinalizerOnlyRemoval checks if the machine update only removes finalizers.
+func isFinalizerOnlyRemoval(m, oldM *machinev1.Machine) (bool, error) {
+	patchBase := client.MergeFrom(oldM)
+	data, err := patchBase.Data(m)
+	if err != nil {
+		return false, fmt.Errorf("cannot calculate patch data from machine object: %w", err)
+	}
+
+	return string(data) == `{"metadata":{"finalizers":[""]}}`, nil
```

Comment:
Still an open question about what happens if there are multiple finalizers, can we do something to make sure that controllers can remove individual finalizers without removing all the finalizers?

Reply:
Could you describe a use case when we need to delete individual finalizers on invalid machines. I think the goal is to allow users to delete machines that were accidentally created with invalid specs. To do so we always need to remove all finalizers, right?

Reply:
I had thought you could only remove one finalizer at a time, but this seems to not be the case, happy to leave as is